### ISSUE TITLE
How to keep connection alive after `onClose` event

### ISSUE THREAD

**Issue by fabiomassimo** [vapor/slack-bot#8]
I started the project from Xcode and everything works as expected, although after some idle time the `onClose` event prints "[CLOSED]" and the application is terminated with exit code: 0.

If I want to deploy this and make sure that the application is always listening for incoming requests should I configure the `WebSocket` properly or is it something I have to specify when deploying?

**Comment by loganwright** [comment]
@fabiomassimo hmmm, I'm not sure, our slackbot has pretty good uptime in terms of weeks more than just moments, is it just inactive when that's happening?

It might be worth setting something up that onClose can wait some time and try to reconnect.

**Comment by fabiomassimo** [comment]
> is it just inactive when that's happening?

Yes, it never occurred while typing or quickly switching between the `bot` and another channel.

> It might be worth setting something up that onClose can wait some time and try to reconnect.

I tried to set a `ping` when the `onClose` event is received or fire it via a cycling timer at the beginning of the application but both solution didn't work out for me. Am I missing a command to force a reconnection?

**Comment by loganwright** [comment]
@fabiomassimo I'm not sure, I'd imagine if we put most of the code in some type of loop that just restarts on a close we'd be ok.

There is a lot of ping/ponging going on in the background intended to keep connection open, perhaps some manual ping additions could be helpful.

Can you tell me what environment you're working in?
